Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/acccerts/DccReissuanceAccCertsFragment.kt

```diff
@@ -0,0 +1,61 @@
+package de.rki.coronawarnapp.dccreissuance.ui.consent.acccerts
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.transition.MaterialSharedAxis
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentDccReissuanceAccCertsBinding
+import de.rki.coronawarnapp.dccreissuance.ui.consent.DccReissuanceAdapter
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.lists.diffutil.update
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
+
+class DccReissuanceAccCertsFragment : Fragment(R.layout.fragment_dcc_reissuance_acc_certs), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val binding: FragmentDccReissuanceAccCertsBinding by viewBinding()
+    private val args by navArgs<DccReissuanceAccCertsFragmentArgs>()
+    private val viewModel: DccReissuanceAccCertsViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as DccReissuanceAccCertsViewModel.Factory
+            factory.create(
+                personIdentifierCode = args.personIdentifierCode,
+            )
+        }
+    )
+
+    private val dccReissuanceAdapter = DccReissuanceAdapter()
```

Comment:
Memory Leak!

The adapter will outlive the view, but holds a reference to the list internally, so the view cannot be destroyed. This even gets worse, when the fragment is kept on the backstack..

Reply:
This is copy paste from the consent view. Can you be more precise? How would the adapter outlive the view? How do you think should it be done?

Reply:
you last change is sufficent.

Reply:
maybe you can change it also in the fragment where you copy pasted it?